### PR TITLE
feat: Attach stacktrace enabled by default 

### PR DESCRIPTION
**Pull request by bruno-garcia**

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Builds hang with this turned on

File: Tests/SentryTests/SentryClientTests.swift [hunk 1]
```diff
@@ -176,7 +176,9 @@ class SentryClientTest: XCTestCase {
     }
     
     func testCaptureErrorWithoutAttachStacktrace() {
-        let eventId = fixture.getSut().capture(error: error, scope: scope)
+        let eventId = fixture.getSut(configureOptions: { options in
+            options.attachStacktrace = true
```
Comment (philipphofmann) [inline Tests/SentryTests/SentryClientTests.swift:180]: `m`: I think this is a small typo here. I wonder why the tests are passing.
```suggestion
            options.attachStacktrace = false
```

File: Tests/SentryTests/SentryClientTests.swift [hunk 2]
```diff
@@ -254,15 +258,15 @@ class SentryClientTest: XCTestCase {
...
-    }
+//
+//    func testScopeIsNotNil() {
+//        let eventId = fixture.getSut().capture(message: message, scope: scope)
+//
+//        eventId.assertIsNotEmpty()
+//        assertLastSentEvent { actual in
+//            XCTAssertEqual(environment, actual.environment)
+//        }
+//    }
```
Comment (philipphofmann) [inline Tests/SentryTests/SentryClientTests.swift:269]: `m`: Can we fix this test here?

**Review by philipphofmann — COMMENTED** [review]
I like this. I made two small comments about the tests.

File: Tests/SentryTests/SentryClientTests.swift (hunk 1 above)
Comment (bruno-garcia) [reply Tests/SentryTests/SentryClientTests.swift:203]: Needs to be reviewed then before merging

**Review by bruno-garcia — COMMENTED** [review]
CI still hangs, not sure why

**Review by philipphofmann — APPROVED** [review]
LGTM

**Comment by bruno-garcia** [comment]
@philipphofmann CI still hangs I think. Not sure we can go on with this change.
Hopefully it's just some issue with CI which we can adddress otherwise I'd be scared even to have this feature available

**Comment by philipphofmann** [comment]
> @philipphofmann CI still hangs I think. Not sure we can go on with this change.
> Hopefully it's just some issue with CI which we can adddress otherwise I'd be scared even to have this feature available

It seems now that the tests are flaky and not the CI hanging. I don't know why `testRateLimitForEvent` failed.

**Comment by philipphofmann** [comment]
I have one idea why the testRateLimitForEvent might have failed. Seems like an async issue in the tests. I'm going to open up a PR for this. @bruno-garcia

**Comment by bruno-garcia** [comment]
Nice!! Thanks for doing this @philipphofmann

**Comment by krugerk** [comment]
https://github.com/getsentry/sentry-cocoa/blob/617cb9ea79d97e833e6263fa862e064c635020d9/Sources/Sentry/Public/SentryOptions.h#L124 appears to indicate that this feature is still disabled by default. From what I can tell, that was the case in the 5.x series but was changed for the 6.x series, as pointed out [here in the migration guide](https://docs.sentry.io/platforms/apple/guides/ios/migration/).

**Comment by bruno-garcia** [comment]
Thanks for pointing it out @krugerk , I've fixed it:
https://github.com/getsentry/sentry-cocoa/commit/d730586df68381256640460cee8d38093f98d85d

**Pull request merged**